modules/oe_theme_helper/src/TwigExtension/TwigExtension.php

```diff
@@ -0,0 +1,488 @@
+    return EuropeanUnionLanguages::getLanguageList();
+  }
+
+  /**
+   * Get date variant class given its status.
+   *
+   * @param string $status
+   *   File extension.
+   *
+   * @return string
+   *   File icon class name.
+   */
+  public static function toDateStatus(string $status): string {
+    $variant_mapping = [
+      'default' => 'default',
+      'ongoing' => 'ongoing',
+      'cancelled' => 'canceled',
+      'past' => 'past',
+    ];
+
+    return array_key_exists($status, $variant_mapping) ? $variant_mapping[$status] : $status;
+  }
+
+  /**
+   * Get file icon class given its extension.
+   *
+   * @param string $extension
+   *   File extension.
+   *
+   * @return string
+   *   File icon class name.
+   */
+  public function toFileIcon(string $extension): string {
+    $extension = strtolower($extension);
+    $extension_mapping = [
+      'image' => [
+        'jpg',
+        'jpeg',
+        'gif',
+        'png',
+        'webp',
+      ],
+      'presentation' => [
+        'ppt',
+        'pptx',
+        'pps',
+        'ppsx',
+        'odp',
+      ],
+      'spreadsheet' => [
+        'xls',
+        'xlsx',
+        'ods',
+      ],
+      'video' => [
+        'mp4',
+        'mov',
+        'mpeg',
+        'avi',
+        'm4v',
+        'webm',
+      ],
+    ];
+
+    foreach ($extension_mapping as $file_type => $extensions) {
+      if (in_array($extension, $extensions)) {
+        return $file_type;
+      }
+    }
+
+    return 'file';
+  }
+
+  /**
+   * Convert Drupal attribute arrays to ECL Twig compatible ones.
+   *
+   * ECL Twig expects an array of string attributes, keyed by a name/value pair.
+   * We use Drupal's Attribute class to make sure that we always get a printable
+   * string, regardless of what we get as input from accessor preprocesses.
+   *
+   * @param mixed $attributes
+   *   Drupal attributes, what we initialize the Drupal's Attribute class with.
+   *
+   * @return array
+   *   An ECL Twig compatible attributes list.
+   */
+  public function toEclAttributes($attributes): array {
+    // Only deal with iterable data.
+    if (!is_iterable($attributes)) {
+      return [];
+    }
+
+    $ecl_attributes = [];
+    $attributes = new Attribute($attributes);
+    foreach ($attributes as $key => $value) {
+      $ecl_attributes[] = [
+        'name' => $key,
+        'value' => (string) $value,
+      ];
+    }
+
+    return $ecl_attributes;
+  }
+
+  /**
+   * Convert icon names to the ECL supported names and apply rotation if needed.
+   *
+   * @param string $icon
+   *   The icon to be converted.
+   *
+   * @return array
+   *   Icon array for ECL components containing icon name, path and rotation.
+   */
+  public function toEclIcon(string $icon): array {
+    // ECL Icons path.
+    $path = base_path() . drupal_get_path('theme', 'oe_theme') .
+      '/dist/' . theme_get_setting('component_library') .
+      '/images/icons/sprites/icons.svg';
```

Comment:
We can request Twig context when declaring the function, after that we can simply access the variable set in `oe_theme_preprocess()` via the context, so that we don't have to duplicate this code here.

```php
class TwigFunction
{
    protected $name;
    protected $callable;
    protected $options;
    protected $arguments = [];

    public function __construct($name, $callable, array $options = [])
    {
        $this->name = $name;
        $this->callable = $callable;
        $this->options = array_merge([
            'needs_environment' => false,
            'needs_context' => false,
            'is_variadic' => false,
            'is_safe' => null,
            'is_safe_callback' => null,
            'node_class' => '\Twig\Node\Expression\FunctionExpression',
            'deprecated' => false,
            'alternative' => null,
        ], $options);
    }
```